### PR TITLE
strengthen typing of goto_program2codet::convert_instruction

### PR DESCRIPTION
**Pull request by kroening**
This enables the use of the correct APIs.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 860e6d5352d342c42c1af702f32143b7bba44c2f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100345397
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: fa7a6e88a93bb9d7b0defd1a2788cb017737aebd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100345615

File: src/goto-instrument/goto_program2code.cpp [hunk 1]
```diff
@@ -178,12 +178,12 @@ goto_programt::const_targett goto_program2codet::convert_instruction(
...
     case DEAD:
       // ignore for now
-      dest.copy_to_operands(code_skipt());
+      dest.add(code_skipt());
       return target;
 
     case FUNCTION_CALL:
     case OTHER:
-      dest.copy_to_operands(target->code);
+      dest.add(target->code);
```
Comment (tautschnig) [inline src/goto-instrument/goto_program2code.cpp]: I am surprised there is still a use of `target->code`, shouldn't this be `target->get_other()`?

**Review by tautschnig — APPROVED** [review]

File: src/goto-instrument/goto_program2code.cpp (hunk 1 above)
Comment (kroening) [reply src/goto-instrument/goto_program2code.cpp]: Indeed, I've added a commit to fix this.

Comment (tautschnig) [reply src/goto-instrument/goto_program2code.cpp]: Thank you!!

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]
LGTM

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 52a3bba3b9a6eca6299874c6c81aa115cf49e97e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100439437

**Pull request merged**